clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5689,10 +5656,8 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       if (!Args.getLastArg(options::OPT_O_Group))
         CmdArgs.push_back("-O2");
 
-      // Default value for FPGA is false, for all other targets is true.
       if (!Args.hasFlag(options::OPT_fsycl_early_optimizations,
-                        options::OPT_fno_sycl_early_optimizations,
-                        !IsFPGASYCLOffloadDevice))
+                        options::OPT_fno_sycl_early_optimizations, true))
```

Comment:
Are we doing 'no early optimizations' for any other target?

Thanks

Reply:
Not that I know of.  But we should keep the logic in case somebody asks for it on the command line.